src/libstd/fs.rs

```diff
@@ -449,6 +449,13 @@ impl Read for File {
         self.inner.read(buf)
     }
 
+    fn size_hint(&self) -> usize {
+        match self.metadata() {
+            Ok(meta) => meta.len() as usize,
```

Comment:
It's a minor point but this should probably use a saturating cast not `as`.

Reply:
Is there an API for doing saturating integer casts? Writing `#[cfg(target_pointer_size = …)]` code here seems tedious.

Reply:
Sadly not yet (rust-lang/rfcs#1218). Something like `cmp::min(meta.len(), usize::max_value() as u64) as usize` should work though.

Reply:
On the other hand this is a hint, the default impl returns zero without necessarily being at EOF so underestimating should be OK. And if you’re reading a file larger than your address space, you’re gonna have a hard time in `read_to_end` anyway.

Reply:
This also needs to check its current position and subtract that off, right?

Reply:
Right. However the obvious fix doesn’t work because `<File as Seek>::seek` takes `&mut self`, even with `SeekFrom::Current(0)`. I wonder if there should be some API like `File::position(&self) -> io::Result<u64>`.

Reply:
Never mind, `<std::fs::File as Seek>::seek` calls `std::sys::fs::File::seek` which takes `&self`, so `size_hint` can call the latter directly.